dev/requirements.txt

```diff
@@ -37,6 +37,12 @@ nbsphinx
 numpydoc
 jinja2
 sphinx==4.5.0
+# With sphinx 4.5.0, we need to pin 'sphinxcontrib-*', it should unpin 'sphinxcontrib-*' after upgrading sphinx>5
+sphinxcontrib-applehelp == 1.0.4
```

Comment:
Can we set upperbound instead? This is for developement (not CI) so we don't really need to pin if it works.

Reply:
I made a check that which versions are compatible with sphinx 4.x and set the upperbound version finally.  See the changelogs here:
https://github.com/sphinx-doc/sphinxcontrib-applehelp/blob/master/CHANGES
https://github.com/sphinx-doc/sphinxcontrib-devhelp/blob/master/CHANGES
https://github.com/sphinx-doc/sphinxcontrib-htmlhelp/blob/master/CHANGES
https://github.com/sphinx-doc/sphinxcontrib-jsmath/blob/master/CHANGES
https://github.com/sphinx-doc/sphinxcontrib-qthelp/blob/master/CHANGES
https://github.com/sphinx-doc/sphinxcontrib-serializinghtml/blob/master/CHANGES

Reply:
Issue title, PR title and description are also updated.